ocs_ci/utility/perf_dash/dashboard_api.py

```diff
@@ -0,0 +1,510 @@
+"""
+API module to interact with the Performance Dashboard
+In order to push results into the dashboard, and pulling data
+from it to compare between results.
+
+"""
+# Builtin modules
+import logging
+
+# 3ed party modules
+import mysql.connector
+
+# Local modules
+from ocs_ci.framework import config
+from ocs_ci.ocs import exceptions
+
+log = logging.getLogger(__name__)
+
+
+class PerfDash(object):
+    """
+    The API class to connect and managing the performance dashboard database
+    """
+
+    def __init__(self):
+        """
+        Initializing the dashboard object and make a connection
+
+        Raise:
+            if credential file can not be open / read
+            if the connection failed
+
+        """
+
+        # Reading connection information and credentials from local file
+        # which is not stored in the GitHub repository
+        self.creds = config.AUTH.get("perf_dashboard", None)
+        if self.creds is None:
+            log.error("Dashboard credentials are not defined in configuration")
+            raise exceptions.MissingRequiredConfigKeyError(
+                "The key AUTH:perf_dashboard is missing in configuration file"
```

Comment:
Not sure if it will be requirement for some tests, but do we have some logic to not fail tests which are using this dashboard if auth data are not provided?
Let's say for IBM guys if they would like to run those tests and they don't have dashboard accessible if it will just not put data there or it will fail the test cases?

Thanks for clarification.

Reply:
in the test itself, we will catch this exception, and if it can not push results to the dashboard it will just print out the results to the log / results_file.

Reply:
Cool